include/deal.II/integrators/laplace.h

```diff
@@ -680,7 +680,7 @@ namespace LocalIntegrators
      * <i>Z<sub>i</sub></i> orthogonal to the current face.
      */
     template <int dim, int spacedim, typename number>
-    double
+    DEAL_II_DEPRECATED_EARLY double
     compute_penalty(const MeshWorker::DoFInfo<dim, spacedim, number> &dinfo1,
                     const MeshWorker::DoFInfo<dim, spacedim, number> &dinfo2,
                     unsigned int                                      deg1,
```

Comment:
To illustrate why it is time for these functions to go:
* In the documentation above, the text starts with "is done in two steps: first, we ..." but the documentation doesn't list what comes next. It's just missing.
* In the implementation, we ask whether one or the other of the two cells adjacent to a face is refined, and then adjust the penalty factor *of the first cell*, rather than of the correct one.

These kinds of issues are just *pervasive* throughout this namespace, and nobody has taken the time or the interest to fix them over the past twenty years. I see no other approach to fixing this stuff than to just nuke it altogether.